pkg/syncer/volume_health.go

```diff
@@ -77,58 +77,75 @@ func csiGetVolumeHealthStatus(ctx context.Context, k8sclient clientset.Interface
 		}
 	}
 
-	for _, vol := range queryResult.Volumes {
-		log.Debugf("Volume %q Health Status %q", vol.VolumeId.Id, vol.HealthStatus)
+	for volID, pvc := range volumeHandleToPvcMap {
+		foundVol := false
+		for _, vol := range queryResult.Volumes {
```

Comment:
If there are "n" PVCs and "m" CNS volumes, the time complexity of the outer for-loop is O(nm). You can reduce it to O(n) if you convert queryResult.Volumes into a hash map.

Reply:
+1

Reply:
Done